packages/types/lib/integration/api.ts

```diff
@@ -10,19 +18,117 @@ export type GetListIntegrations = Endpoint<{
         }[];
     };
 }>;
-
 export type DeleteIntegrationPublic = Endpoint<{
     Method: 'DELETE';
     Path: '/config/:providerConfigKey';
     Params: { providerConfigKey: string };
     Success: { success: true };
 }>;
 
+export type PostIntegration = Endpoint<{
+    Method: 'POST';
+    Path: '/api/v1/integrations';
+    Body: { provider: string };
+    Success: {
+        data: ApiIntegration;
+    };
+}>;
+
+export type ApiIntegration = Merge<IntegrationConfig, ApiTimestamps>;
+export type GetIntegration = Endpoint<{
+    Method: 'GET';
+    Path: '/api/v1/integrations/:providerConfigKey';
+    Params: { providerConfigKey: string };
+    Success: {
+        data: {
+            integration: ApiIntegration;
+            template: Template;
+            meta: {
+                connectionsCount: number;
+                webhookUrl: string | null;
+                webhookSecret: string | null;
+            };
+        };
+    };
+}>;
+
+export type PatchIntegration = Endpoint<{
+    Method: 'PATCH';
+    Path: '/api/v1/integrations/:providerConfigKey';
+    Params: { providerConfigKey: string };
+    Body:
+        | { integrationId?: string | undefined }
+        | {
+              authType: Extract<AuthModeType, 'OAUTH1' | 'OAUTH2' | 'TBA'>;
```

Comment:
OAUTH2_CC?

Reply:
I'm following what is currently in the ui and oauth2_cc doesn't have any credentials attached in the UI (e.g: checkout) is that a mistake?

Reply:
Gotcha, this is fine then, thanks for clarifying